server/src/main/java/org/elasticsearch/index/codec/tsdb/es819/ES819TSDBDocValuesProducer.java

```diff
@@ -1941,4 +1952,46 @@ public void close() {
         }
     }
 
+    static final class SingletonLongToSingletonIntDelegate implements BlockLoader.SingletonLongBuilder {
```

Comment:
Ideally I don't see this somewhat repeated logic, but I don't see a way around it.